src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/SignedXml.cs

```diff
@@ -809,33 +812,33 @@ private byte[] GetC14NDigest(HashAlgorithm hash)
 
         private int GetReferenceLevel(int index, ArrayList references)
         {
-            if (_refProcessed[index]) return _refLevelCache[index];
+            if (_refProcessed![index]) return _refLevelCache![index];
```

Comment:
A whole bunch of `!`s can be removed from this method by simply starting it with

```C#
Debug.Assert(_refProcessed != null);
Debug.Assert(_refLevelCache != null);
```

Reply:
Thanks - I didn't realise that that acted as a hint to the null analyzer!